samples: matter: Added mode for switching Thread and Wi-Fi (#9707)
nrfconnect/sdk-nrf: samples/matter/lock/README.rst

      * If pressed for less than three seconds, it changes the lock state to the opposite one.
      * If pressed for more than three seconds, it starts the NFC tag emulation, enables Bluetooth LE advertising for the predefined period of time (15 minutes by default), and makes the device discoverable over Bluetooth LE.

Button 3:

>     * On nRF5340 DK with using `thread_wifi_switched` build type: If pressed for more than ten seconds, it swaps the running application from Thread to Wi-Fi or opposite and factory resets the device.

Reviewer: ```suggestion
    * On the nRF5340 DK when using the ``thread_wifi_switched`` build type: If pressed for more than ten seconds, it switches the running application from Thread or Wi-Fi to the other and factory resets the device.
```

>     * On other: Not available.

Reviewer: ```suggestion
    * On other platform or build type: Not available.
```
